pkg/applications/helmclient/client.go

```diff
@@ -0,0 +1,332 @@
+	}
+
+	var out strings.Builder
+	man := &downloader.Manager{
+		Out:              &out,
+		ChartPath:        chartLoc,
+		Getters:          h.getterProviders,
+		RepositoryConfig: h.settings.RepositoryConfig,
+		RepositoryCache:  h.settings.RepositoryCache,
+		RegistryClient:   regClient,
+		Debug:            true,
+		Verify:           downloader.VerifyNever,
+		SkipUpdate:       true,
+		// todo credentials
+	}
+
+	chartToInstall, err := loader.Load(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not load chart: %w", err)
+	}
+
+	fi, err := os.Stat(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not find chart at `%s': %w", chartLoc, err)
+	}
+
+	// If we got the chart from the filesystem (i.e. cloned from a git repository), we have to build dependencies because
+	// charts directory may not exist.
+	//
+	// note: if we got the chart from a remote helm repository, we don't have to build dependencies because the package
+	// (i.e. the tgz) should already contain it.
+	if fi.IsDir() {
+		// Helm does not download dependency if the repository is unknown (i.e. not present in repository.xml)
+		// so we explicitly add to the repository file.
+		var dependencies []*chart.Dependency
+		if chartToInstall.Lock != nil {
+			dependencies = chartToInstall.Lock.Dependencies
+		} else {
+			dependencies = chartToInstall.Metadata.Dependencies
+		}
+
+		for _, dep := range dependencies {
+			// oci or file dependencies can not be added as a repository.
+			if strings.HasPrefix(dep.Repository, "http://") || strings.HasPrefix(dep.Repository, "https://") {
+				if _, err := h.ensureRepository(dep.Repository); err != nil {
+					return nil, fmt.Errorf("can not download index for repository: %w", err)
+				}
+			}
+		}
+
+		// Equivalent of helm dependency build.
+		err = man.Build()
+		if err != nil {
+			h.logger.Errorw("can not build dependencies", "chart", chartLoc, "log", out.String())
+			return nil, fmt.Errorf("can not build dependencies: %w", err)
+		}
+
+		// We have to reload the chart to load the downloaded dependencies.
+		chartToInstall, err = loader.Load(chartLoc)
+		if err != nil {
+			return nil, fmt.Errorf("can not reload chart: %w", err)
+		}
+	}
+	h.logger.Debugw("successfully built dependencies", "chart", chartLoc, "log", out.String())
+	return chartToInstall, nil
+}
+
+// ensureRepository adds the repository url if it doesn't exist and downloads the latest index file.
+// The repository is added with the name helm-manager-$(sha256 url).
+func (h HelmClient) ensureRepository(url string) (string, error) {
+	repoFile, err := repo.LoadFile(h.settings.RepositoryConfig)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return "", err
+	}
+
+	repoName, err := computeRepoName(url)
+	if err != nil {
+		return "", fmt.Errorf("can not compute repository's name for '%s': %w", url, err)
+	}
+	desiredEntry := &repo.Entry{
+		Name: repoName,
+		URL:  url,
+		// todo credential  ???
```

Comment:
If I read the helm code correctly, we could handle the complete auth in the getters themselves:
Example for HTTP:
* HTTP Getter can be created with getter Options set https://github.com/helm/helm/blob/main/pkg/getter/httpgetter.go#L101
* and then any form of credentials can be set like BasicAuth, TLSClientConfig, ... https://github.com/helm/helm/blob/657850e44b880cca43d0606ebf5a54eb75362c3f/pkg/getter/getter.go#L63 And then that should be passed along every request with 

And since funcs like `NewChartRepository` also always pass the getters along, that should work. 
I think the big advantage would be that we could support both HTTP as well as OCI Getters

Reply:
i think so too... I removed the it :)